ionoscloud/resource_group.go

```diff
@@ -110,18 +117,51 @@ func resourceGroup() *schema.Resource {
 				},
 			},
 		},
+		Timeouts:      &resourceDefaultTimeouts,
+		SchemaVersion: 1,
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Type:    resourceGroup0().CoreConfigSchema().ImpliedType(),
+				Upgrade: resourceGroupUpgradeV0,
+				Version: 0,
+			},
+		},
+	}
+}
+
+func resourceGroup0() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
 		Timeouts: &resourceDefaultTimeouts,
 	}
 }
 
+func resourceGroupUpgradeV0(_ context.Context, state map[string]interface{}, _ interface{}) (map[string]interface{}, error) {
+	oldState := state
+	var oldData string
+	if d, ok := oldState["user_id"].(string); ok {
+		oldData = d
+	}
+
+	var users []string
```

Comment:
should this block be inside the if statement above, or do we want this to run anyway?